### DISCUSSION TITLE
FCM Push notification's on IOS - pushNotificationReceived listener not called

### DISCUSSION THREAD

**Discussion by dave-nelson-uk** [ionic-team/capacitor#4601] [Q&A]
Hi

We're utilising FCM push notifications with our app for both Android and IOS. Everything works perfectly on Android, but we're seeing issues for IOS.

Notifications are being sent correctly from our server's, and being received by the device as they are appearing in the IOS notification list when the app is not open, but they are not activating the pushNotificationReceived event. The listeners are in place.
Example : 

```
 // Show us the notification payload if the app is open on our device
        PushNotifications.addListener('pushNotificationReceived',
            (notification: PushNotification) => {
                alert("PUSH NOTIFICATION")
                //  alert('Push received: ' + JSON.stringify(notification));
            }
        );
```

Registration is in place : 

```
PushNotifications.requestPermission().then(result => {
            if (result.granted) {
                // Register with Apple / Google to receive push via APNS/FCM
                PushNotifications.register();
            } else {
                // Show some error
                this.finishLogin();
            }
        });
```
 

AppDelegate.swift changes are in place:

```
import UIKit
import Capacitor
import Firebase
import FirebaseCore
import FirebaseInstanceID
import FirebaseMessaging

@UIApplicationMain
class AppDelegate: UIResponder, UIApplicationDelegate {

  var window: UIWindow?


  func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
    // Override point for customization after application launch.
    FirebaseApp.configure()
    return true
  }

```

```
func application(_ application: UIApplication, didRegisterForRemoteNotificationsWithDeviceToken deviceToken: Data) {
    Messaging.messaging().apnsToken = deviceToken
    Messaging.messaging().token(completion: {(token, error) in
        if let error = error {
            NotificationCenter.default.post(name:Notification.Name(CAPNotifications.DidFailToRegisterForRemoteNotificationsWithError.name()), object:error)
        }else if let token = token {
            NotificationCenter.default.post(name:Notification.Name(CAPNotifications.DidRegisterForRemoteNotificationsWithDeviceToken.name()),object: token)
        }
    })
  
  }

  func application(_ application: UIApplication, didFailToRegisterForRemoteNotificationsWithError error: Error) {
    NotificationCenter.default.post(name: Notification.Name(CAPNotifications.DidFailToRegisterForRemoteNotificationsWithError.name()), object: error)
  }

```

Podfile looks like this :

```
import UIKit
import Capacitor
import Firebase
import FirebaseCore
import FirebaseInstanceID
import FirebaseMessaging

@UIApplicationMain
class AppDelegate: UIResponder, UIApplicationDelegate {

  var window: UIWindow?


  func application(_ application: UIApplication, didFinishLaunchingWithOptions launchOptions: [UIApplication.LaunchOptionsKey: Any]?) -> Bool {
    // Override point for customization after application launch.
    FirebaseApp.configure()
    return true
  }
```

I dont think any of the plugins I have would conflict either:

D:\youin\youin>npx cap copy
√ Copying web assets from www to android\app\src\main\assets\public in 603.82ms
√ Copying native bridge in 1.02ms
√ Copying capacitor.config.json in 1.10ms
  Found 6 Cordova plugins for android
    cordova-plugin-badge (0.8.8)
    cordova-plugin-device (2.0.3)
    cordova-plugin-file (6.0.2)
    cordova-plugin-geolocation (4.1.0)
    cordova-plugin-local-notification (0.9.0-beta.2)
    cordova-plugin-purchase (10.5.3)
√ copy in 748.28ms
√ Copying web assets from www to ios\App\public in 526.54ms
√ Copying native bridge in 932.30μp
√ Copying capacitor.config.json in 1.10ms
  Found 6 Cordova plugins for ios
    cordova-plugin-badge (0.8.8)
    cordova-plugin-device (2.0.3)
    cordova-plugin-file (6.0.2)
    cordova-plugin-geolocation (4.1.0)
    cordova-plugin-local-notification (0.9.0-beta.2)
    cordova-plugin-purchase (10.5.3)
√ copy in 574.69ms
√ copy in 278.00μp

Any clues as to why its not working, or is this a bug?

**Comment by jcesarmobile** [comment]
apps can only have one delegate for the notifications, and `cordova-plugin-local-notification` is probably setting it's own and stopping capacitor's delegate from working
try removing that plugin
